web_sdk/BUILD.gn

```diff
@@ -559,7 +559,8 @@ if (!is_fuchsia) {
            ] + web_engine_libraries
 
     if (build_canvaskit) {
-      deps += [ "//third_party/skia/modules/canvaskit" ]
+      deps +=
```

Comment:
I would need to investigate more to make specific suggestions about how to guard the canvaskit toolchain definitions with visibility annotations or asserts, but one thing that will probably help in any case is putting the targets you want to build with those toolchains right next to the toolchain definitions, with comments saying not to use those toolchains for anything else.

So here, you'd depend on a `group` target, where the `group` has a public_deps on `//third_party/skia/...`, and the `group` target is in the same file as the toolchain definitions.